gapic/templates/%namespace/%name_%version/%sub/services/%service/_client_macros.j2

```diff
@@ -36,7 +36,7 @@
             {% elif not method.server_streaming %}
             ) -> {{ method.client_output.ident }}:
             {% else %}
-            ) -> Iterable[{{ method.client_output.ident }}]:
+            ) -> Union[Iterable[{{ method.client_output.ident }}], GrpcStream[{{ method.client_output.ident }}]]:
```

Comment:
Did you find a solution for how to deal with `api_core.retry.Retry` in this?

The retry object wraps the rpc iterator, and it doesn't currently expose `grc.Call` methods, so anything involving retires won't return a GrpcStream. It's not exactly clear to me how we should expose metadata at the retry level (do we want to give metadata for each failed call, or only the final successful one? I think each failed request could be needed?)

I don't think this is relevant for bigtable, since we extend the retry logic. But it's probably needed in other contexts

Reply:
We could consider adding this as future work, given that it's more than just changing the response type.

Reply:
Ok, sounds good

FWIW, I think it would be a good idea to still design a solution for failed attempts, so we don't paint ourselves into a corner here. (maybe we can use the retry.on_error callback for this?) But yeah, I'll assume we just want one set of metadata for now